app/src/main/java/org/mozilla/focus/repository/BookmarkRepository.java

```diff
@@ -0,0 +1,62 @@
+package org.mozilla.focus.repository;
+
+import android.arch.lifecycle.LiveData;
+
+import org.mozilla.focus.persistence.BookmarkModel;
+import org.mozilla.focus.persistence.BookmarksDatabase;
+import org.mozilla.focus.utils.ThreadUtils;
+
+import java.util.List;
+import java.util.UUID;
+
+public class BookmarkRepository {
+    private static volatile BookmarkRepository instance;
+
+    private BookmarksDatabase bookmarksDatabase;
+
+    private BookmarkRepository(final BookmarksDatabase database) {
+        bookmarksDatabase = database;
+    }
+
+    public static BookmarkRepository getInstance(final BookmarksDatabase database) {
+        if (instance == null) {
+            synchronized (BookmarkRepository.class) {
+                if (instance == null) {
+                    instance = new BookmarkRepository(database);
+                }
+            }
+        }
+        return instance;
+    }
+
+    public LiveData<List<BookmarkModel>> loadBookmarks() {
+        return bookmarksDatabase.bookmarkDao().loadBookmarks();
+    }
+
+    public LiveData<BookmarkModel> getBookmarkById(String id) {
+        return bookmarksDatabase.bookmarkDao().getBookmarkById(id);
+    }
+
+    public LiveData<List<BookmarkModel>> getBookmarksByUrl(String url) {
+        return bookmarksDatabase.bookmarkDao().getBookmarksByUrl(url);
+    }
+
+    public String addBookmark(String title, String url) {
+        final BookmarkModel bookmark = new BookmarkModel(UUID.randomUUID().toString(), title, url);
```

Comment:
Just curious: Is the reason why we're passing String rather than ID here because it's more efficient? Is it that an UUID is a heavier object, or is it a Serializable takes longer to serialize/unserialize?